File: app/src/main/java/ru/garretech/garred/doramatv/tools/ListConverter.java
package ru.garretech.garred.doramatv.tools;

import android.arch.persistence.room.TypeConverter;

import java.util.Arrays;
import java.util.List;

public class ListConverter {
    @TypeConverter
    public String fromList(List<String> list) {
        return list.toString();
    }

    @TypeConverter
    public List<String> toList(String listAsString) {
        return Arrays.asList(listAsString.split("\\s*,\\s*"));
    }

}

File: app/src/main/java/ru/garretech/garred/doramatv/tools/ListConverter.kt
package ru.garretech.garred.doramatv.tools

import androidx.room.TypeConverter

class ListConverter {
    @TypeConverter
    fun fromList(list: List<String>): String {
        val newString = list.toString().substring(1,list.toString().length-1)
        return newString
    }

    @TypeConverter
    fun toList(listAsString: String): List<String> {
        val newList = listOf(*listAsString.split("\\s*,\\s*".toRegex()).dropLastWhile { it.isEmpty() }
            .toTypedArray())
        return newList
    }

}
